pilot/pkg/serviceregistry/serviceentry/conversion.go

```diff
@@ -110,12 +129,13 @@ func ServiceToServiceEntry(svc *model.Service, proxy *model.Proxy) *config.Confi
 
 	// Port is mapped from ServicePort
 	for _, p := range svc.Ports {
+		targetPort := getTargetPortFromServiceInstances(p, svc, proxy.ServiceInstances)
```

Comment:
This is not right. As the serviceinstance here is only this proxie's.  May not be the service's

Reply:
u r right, the svc should be imported across all egress listeners. Thanks!

Reply:
Hi @hzxuzhonghu, new submit for picking the service instances from PushRequest. Thanks!